signer/src/wsts_state_machine.rs

```diff
@@ -30,33 +39,435 @@ use wsts::traits::Signer as _;
+pub trait FromMessage {
+    /// Convert the given message into the implementing type.
+    fn from_message(message: &Message) -> Self
+    where
+        Self: Sized;
+}
+
+impl FromMessage for Packet {
+    fn from_message(message: &Message) -> Self {
+        Packet {
+            msg: message.clone(),
+            sig: Vec::new(),
+        }
+    }
+}
+
+/// Wrapper for a WSTS FIRE coordinator state machine.
+#[derive(Debug, Clone, PartialEq)]
+pub struct FireCoordinator(fire::Coordinator<Aggregator>);
+
+impl std::ops::Deref for FireCoordinator {
+    type Target = fire::Coordinator<Aggregator>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for FireCoordinator {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Wrapper for a WSTS FROST coordinator state machine.
+#[derive(Debug, Clone, PartialEq)]
+pub struct FrostCoordinator(frost::Coordinator<Aggregator>);
+
+impl std::ops::Deref for FrostCoordinator {
+    type Target = frost::Coordinator<Aggregator>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for FrostCoordinator {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// A trait for WSTS state machines.
+pub trait WstsCoordinator
+where
+    Self: Sized,
+{
+    /// Creates a new coordinator state machine.
+    fn new<I>(signers: I, threshold: u16, message_private_key: PrivateKey) -> Self
+    where
+        I: IntoIterator<Item = PublicKey>;
+
+    /// Gets the coordinator configuration.
+    fn get_config(&self) -> Config;
+
+    /// Save the state required to reconstruct the state machine.
+    fn save(&self) -> SavedState;
+
+    /// Create a new coordinator state machine from the given aggregate
+    /// key.
+    ///
+    /// # Notes
+    ///
+    /// The `WstsCoordinator` is a state machine that is responsible for
+    /// DKG and for facilitating signing rounds. When created the
+    /// `WstsCoordinator` state machine starts off in the `IDLE` state,
+    /// where you can either start a signing round or start DKG. This
+    /// function is for loading the state with the assumption that DKG has
+    /// already been successfully completed.
+    fn load<S>(
+        storage: &S,
+        aggregate_key: PublicKeyXOnly,
+        signer_public_keys: impl IntoIterator<Item = PublicKey> + Send,
+        threshold: u16,
+        signer_private_key: PrivateKey,
+    ) -> impl Future<Output = Result<Self, error::Error>> + Send
+    where
+        S: storage::DbRead + Send + Sync;
+
+    /// Process the given message.
+    fn process_message(
+        &mut self,
+        message: &Message,
+    ) -> Result<(Option<Packet>, Option<OperationResult>), Error> {
+        let packet = Packet::from_message(message);
+        self.process_packet(&packet)
+    }
+
+    /// Process inbound messages
+    fn process_inbound_messages(
```

Comment:
Is this used around?

Reply:
Nope, apparently only for the `SignerStateMachine` and then only in tests, so the coordinator's don't use the interator variants. Removed in [cb95417](https://github.com/stacks-network/sbtc/pull/1287/commits/cb9541747dc90d6fc486f3308536ebddc830b112)